onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -486,7 +519,8 @@ class PlannerImpl {
         auto current = Index(node_output->Name());
         AllocPlan(current).value_type = utils::GetMLDataType(*node_output);
         MLValueIndex reused;
-        if (std::find(graph_outputs.begin(), graph_outputs.end(), node_output) != graph_outputs.end()) {
+        if (std::find(graph_outputs.begin(), graph_outputs.end(), node_output) != graph_outputs.end() &&
+            !FindReusableGraphInput(*pnode, output_arg_num, &reused)) {
```

Comment:
I don't understand this fallthrough logic, because the lines below will invoke "FindReusableInput" again. (In this case, correctness depends on FindReusableGraph & FindReusableInput "finding" the same "reused".) I think it will be better to use a nested "if FindReusableGraphInput(…)" and do the right thing inside this if-statement than cascading to subsequent else-branches.

Reply:
Hi, How about this comment? Can we change this to: if (this is a graph output) { if (FindReusableGraphInput(…)) Reuse(reused, current) else ...kAllocateOutput" ?